File: countWords_ws/src/count/src/count_client.cpp
#include <ros/ros.h>
#include <count/cont.h>


int main(int argc, char** argv)
{
    ros::init(argc, argv, "client");

    if (argc != 2) {
        ROS_ERROR("Usage: count_client <input_text>");

        return 1;
    }

    ros::NodeHandle nh;
    ros::ServiceClient client = nh.serviceClient<count::cont>("Words_counter");
    count::cont srv;

    srv.request.words = argv[1];
    if (client.call(srv)) {
        ROS_INFO("Response : %d", srv.response.count);
    }
    else {
        ROS_ERROR("Failed to call service words_counter");
        return 1;
    }
    return 0;
}

File: countWords_ws/src/count/scripts/count_client.py
#!/usr/bin/env python3

import rospy
from count.srv import cont
import sys

def client():
    rospy.wait_for_service('word_count')
    try:
        rospy.init_node('client')
        word_counter = rospy.ServiceProxy('word_count', cont)
        words = ''.join(sys.argv[1:])
        words_count = word_counter(words)

        print(words, '->', words_count.count)
        return words_count

    except rospy.ServiceException as E:
        print("Service call failed: %s"%E)


if __name__=="__main__":
    client()